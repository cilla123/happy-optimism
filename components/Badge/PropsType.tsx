export default interface PropsType {
  theme?: 'default' | 'primary' | 'success' | 'warning' | 'error';
  shape?: 'dot' | 'radius' | 'round' | 'circle';
  sup?: boolean;
  text?: any;
}
