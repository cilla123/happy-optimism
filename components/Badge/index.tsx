import React, { PureComponent } from 'react';
import classnames from 'classnames';
import PropsType from './PropsType';

export interface BadgeProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class Badge extends PureComponent<BadgeProps, {}> {

  static defaultProps = {
    prefixCls: 'za-badge',
    theme: 'error',
    sup: false,
  };

  render() {
    const { prefixCls, className, theme, shape, sup, text, children, ...others } = this.props;

    const cls = classnames(`${prefixCls}`, className, {
      [`theme-${theme}`]: !!theme,
      [`shape-${shape}`]: !!shape,
    });

    const supCls = classnames(`${prefixCls}-sup`, {
      [`${prefixCls}-sup-up`]: sup,
    });

    return (
      <span className={cls}>
        {children}
        <sup className={supCls} {...others}>{text}</sup>
      </span>
    );
  }
}
