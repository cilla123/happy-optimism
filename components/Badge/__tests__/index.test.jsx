import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Badge from '../index';

describe('Badge', () => {
  it('renders shape is dot correctly', () => {
    const wrapper = render(<Badge shape="dot" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
