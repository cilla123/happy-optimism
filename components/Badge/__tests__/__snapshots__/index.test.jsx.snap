// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders shape is dot correctly 1`] = `
<span
  class="za-badge theme-error shape-dot"
>
  <sup
    class="za-badge-sup"
  />
</span>
`;
